java/runtime/src/main/java/org/ray/runtime/runner/RunManager.java

```diff
@@ -92,9 +102,19 @@ public void terminateProcess(String name, Process p) {
 
   private void createTempDirs() {
     try {
-      FileUtils.forceMkdir(new File(rayConfig.logDir));
-      FileUtils.forceMkdir(new File(rayConfig.rayletSocketName).getParentFile());
-      FileUtils.forceMkdir(new File(rayConfig.objectStoreSocketName).getParentFile());
+      FileUtils.forceMkdir(new File(sessionDir));
+      if (StringUtils.isNullOrEmpty(rayConfig.getLogDir())) {
+        rayConfig.setLogDir(String.format("%s/logs", sessionDir));
+      }
+      if (StringUtils.isNullOrEmpty(rayConfig.getRayletSocketName())) {
+        rayConfig.setRayletSocketName(String.format("%s/sockets/raylet", sessionDir));
+      }
+      if (StringUtils.isNullOrEmpty(rayConfig.getObjectStoreSocketName())) {
+        rayConfig.setObjectStoreSocketName(String.format("%s/sockets/object_store", sessionDir));
+      }
```

Comment:
better set the above configs in `RayConfig`'s constructor. so if we call `rayConfig.getLogDir()` before this, it still returns the correct value.